src/Serializer.php

```diff
@@ -16,7 +16,7 @@ final class Serializer extends BaseSerializer
     private const KEY_TYPE = '#type';
     private const KEY_SCALAR = '#scalar';
 
-    public function normalize($data, $format = null, array $context = [])
+    public function normalize($data, $format = null, array $context = []): \ArrayObject|array|string|int|float|bool|null
```

Comment:
That's a BC break for people still using PHP 7.4

Reply:
I suggest to just bump the minimal PHP version.

Reply:
I've introduced a check for the php version if you'd like to keep php 7.1 compatibility but honestly might as well get rid of old stuff and tag it with a new major version, just let me know which solution you prefer